File: py/1219.py
while True:
    try:
        a, b, c = [int(i) for i in input().split()]
        PI = 3.1415926535897
        p = (a + b + c) / 2

        triangleArea = (p * (p - a) * (p - b) * (p - c)) ** 0.5
        innerCircleRadius = triangleArea / p
        innerCircleArea = PI * (innerCircleRadius ** 2)
        outerCircleRadius = (a * b * c) / (4 * triangleArea)
        outerCircleArea =   PI * (outerCircleRadius ** 2)

        sunflowers = outerCircleArea - triangleArea
        violets = triangleArea - innerCircleArea
        roses = innerCircleArea

        print(f'{(sunflowers):.4f} {(violets):.4f} {(roses):.4f}')

    except EOFError:
        break

File: cpp/1219.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    int a, b, c;
    const double PI = 3.1415926535897;

    while (scanf("%d %d %d", &a, &b, &c) != EOF) {
        double p = (double)(a + b + c) / 2;

        double triangleArea = sqrt(p * (p - a) * (p - b) * (p - c));
        double innerCircleRadius = triangleArea / p;
        double innerCircleArea = PI * pow(innerCircleRadius, 2);
        double outerCircleRadius = (a * b * c) / (4 * triangleArea);
        double outerCircleArea = PI * pow(outerCircleRadius, 2);

        double sunflowers = outerCircleArea - triangleArea;
        double violets = triangleArea - innerCircleArea;
        double roses = innerCircleArea;

        printf("%.4f %.4f %.4f\n", sunflowers, violets, roses);
    }

    return 0;
}